manager/allocator/network.go

```diff
@@ -81,6 +81,7 @@ func (a *Allocator) doNetworkInit(ctx context.Context) error {
 		unallocatedNetworks: make(map[string]*api.Network),
 		ingressNetwork:      newIngressNetwork(),
 	}
+	a.netCtx = nc
```

Comment:
The reason `a.netCtx` is initialized at the end is to make sure in case of failures we don't return from `doNetworkInit` with a `netCtx` which shouldn't be there in the `Allocator`. This is because the `Allocator` has a longer life time than what happens in `doNetworkInit` itself. 

So in all of `doNetworkInit` I've just passed the `netCtx` as an argument to functions that need it. May be we should do the same for `taskCreateNetworkAttachments` i.e add `netCtx` as an argument?